File: 13-10-2025_Find Resultant Array After Removing Anagrams.java
import java.util.*;

class Solution {
    public List<String> removeAnagrams(String[] words) {
        List<String> ans = new ArrayList<>();
        ans.add(words[0]);
        
        for (int i = 1; i < words.length; i++) {
            String a = words[i];
            String b = ans.get(ans.size() - 1);
            
            char[] aChars = a.toCharArray();
            char[] bChars = b.toCharArray();
            Arrays.sort(aChars);
            Arrays.sort(bChars);
            
            if (!Arrays.equals(aChars, bChars)) {
                ans.add(words[i]);
            }
        }
        return ans;
    }
}

File: 13-10-2025_Find Resultant Array After Removing Anagrams.py
class Solution:
    def removeAnagrams(self, words: List[str]) -> List[str]:
        ans = [words[0]]
        
        for i in range(1, len(words)):
            a_sorted = ''.join(sorted(words[i]))
            b_sorted = ''.join(sorted(ans[-1]))
            
            if a_sorted != b_sorted:
                ans.append(words[i])
                
        return ans